compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -1173,7 +1173,8 @@ fn quoted_tt_to_string(tt: &mbe::TokenTree) -> String {
         mbe::TokenTree::MetaVar(_, name) => format!("${}", name),
         mbe::TokenTree::MetaVarDecl(_, name, kind) => format!("${}:{}", name, kind),
         _ => panic!(
-            "unexpected mbe::TokenTree::{{Sequence or Delimited}} \
+            "{}",
+            "unexpected mbe::TokenTree::{Sequence or Delimited} \
              in follow set checker"
         ),
```

Comment:
I would have imagined that this case would be accepted as correct: the panic message is explicitly escaping the braces.

Reply:
It'd be properly escaped if it was used as a formatting argument, yes. The problem is that it isn't. With `"{{"` it literally outputs `"{{"`. `panic!()` does not use its argument as a format string if there's no additional arguments. That's why I want to add this lint, so it can be fixed in Rust 2021 while giving warnings about incompatible uses (and mistakes) in Rust 2018/2015.

Reply:
Oh! I see. Makes sense.